components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/MyAccountAuthorizedAPIListener.java

```diff
@@ -108,6 +117,9 @@ public void postGetAuthorizedAPIs(List<AuthorizedAPI> authorizedAPIList, String
                     if (!authorizedNoPolicyAPIIdentifiers.contains(systemAPIResource.getIdentifier())) {
```

Comment:
if (!authorizedNoPolicyAPIIdentifiers.contains(systemAPIResource.getIdentifier()) && !authorizedRBACAPIIdentifiers.contains(systemAPIResource.getIdentifier())) {
  continue;
}

Otherwise impersonation related scopes will be skipped from the first if block.